ddtrace/appsec/processor.py

```diff
@@ -210,9 +217,51 @@ def __attrs_post_init__(self):
         # we always need the response headers
         self._mark_needed(_Addresses.SERVER_RESPONSE_HEADERS_NO_COOKIES)
 
+    def update_rules(self, new_rules):
+        # type: (List[Dict[str, Any]]) -> None
+        try:
+            self._ddwaf.update_rules(new_rules)
+        except TypeError as e:
+            log.debug("Error updating RCM rules: %s", str(e))  # noqa: G200
+
     def on_span_start(self, span):
         # type: (Span) -> None
-        pass
+
+        peer_ip = _asm_context.get_ip()
+        headers = _asm_context.get_headers()
+        headers_case_sensitive = _asm_context.get_headers_case_sensitive()
+
+        if headers is not None:
+            _context.set_items(
+                {
+                    "http.request.headers": headers,
+                    "http.request.headers_case_sensitive": headers_case_sensitive,
+                },
+                span=span,
+            )
+
+            if peer_ip is not None:
+                ip = trace_utils._get_request_header_client_ip(headers, peer_ip, headers_case_sensitive)
+                # Save the IP and headers in the context so the retrieval can be skipped later
+                _context.set_item("http.request.remote_ip", ip, span=span)
+                if ip and self._is_needed(_Addresses.HTTP_CLIENT_IP):
+                    data = {_Addresses.HTTP_CLIENT_IP: ip}
+                    log.debug("[DDAS-001-00] Executing AppSec In-App WAF with parameters: %s", data)
+                    ddwaf_result = self._run_ddwaf(data)
+
+                    if ddwaf_result and ddwaf_result.actions and "block" in ddwaf_result.actions and ddwaf_result.data:
+                        res_dict = json.loads(ddwaf_result.data)
+                        log.debug("[DDAS-011-00] AppSec In-App WAF returned: %s", res_dict)
```

Comment:
```suggestion
                        log.debug("[DDAS-011-00] ASM WAF returned: %s", res_dict)
```

will there be sensitive data in this return value?

Reply:
I think this was talked some time ago on some Slack ASM channel and it was decided that it was fine if the loglevel was debug but I'll ask again tomorrow just to be sure since slack deleted already that conversation.

Reply:
:+1: thanks and apologies if this was already something long decided and I'm digging it back up 😄